e2e/testcases/remediator_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/metrics"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	"kpt.dev/configsync/pkg/api/configmanagement"
+	"kpt.dev/configsync/pkg/api/configsync"
+	"kpt.dev/configsync/pkg/core"
+	"kpt.dev/configsync/pkg/status"
+	"kpt.dev/configsync/pkg/testing/fake"
+)
+
+func TestSurfaceFightError(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.DriftControl)
+
+	nt.T.Logf("Stop the admission webhook to generate the fights")
+	nomostest.StopWebhook(nt)
+
+	ns := fake.NamespaceObject("test-ns", core.Annotation("foo", "bar"))
+	rb := roleBinding("test-rb", ns.Name, map[string]string{"foo": "bar"})
+	nt.RootRepos[configsync.RootSyncName].Add(
+		fmt.Sprintf("acme/namespaces/%s/ns.yaml", ns.Name), ns)
+	nt.RootRepos[configsync.RootSyncName].Add(
+		fmt.Sprintf("acme/namespaces/%s/rb.yaml", ns.Name), rb)
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add Namespace and RoleBinding")
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// Make the # of updates exceed the fightThreshold defined in pkg/syncer/reconcile/fight_detector.go
+	go func() {
+		for i := 0; i <= 5; i++ {
```

Comment:
Is this going to be flakey if CS misses any of these events?

Reply:
You mean if the patch fails? If it happens, I think the test will fail with `nt.T.Fatal(err)`.

Reply:
No, I mean if the patch succeeds, but the remediator watch misses the event. I think this would only happen if the reconciler is rescheduled, but that can happen on autopilot clusters when the node pool is changed.

I don't know a good way to defend against it tho, so maybe it doesn't matter.